### PR TITLE
Add `Alias` hooks and tests

### PR DESCRIPTION
**Pull request by RedbackThomson**
Description of changes:
- Rename `AliasName` to `Name` to reduce redundancy
- Add hooks for proper `ReadMany` input and parsing output
- Add test for `Alias`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: templates/hooks/alias/sdk_read_many_pre_set_output.go.tpl [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+for _, elem := range resp.Aliases {
+  if elem.AliasName == nil || r.ko.Spec.Name == nil {
+    continue
+  }
+  
+  if *elem.AliasName == *r.ko.Spec.Name {
+    matchingAliases = append(matchingAliases, elem)
+  }
+}
+resp.Aliases = matchingAliases
```
Comment (jaypipes) [inline templates/hooks/alias/sdk_read_many_pre_set_output.go.tpl:12]: for the template code, you may consider using TAB characters and indenting the code to the right level. I know that `go fmt` cleans all that up when we run `make build-controller` so not a huge deal.

File: test/e2e/resources/alias_simple.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+apiVersion: kms.services.k8s.aws/v1alpha1
+kind: Alias
+metadata:
+  name: $ALIAS_NAME
+spec:
+  name: alias/$ALIAS_NAME
```
Comment (jaypipes) [inline test/e2e/resources/alias_simple.yaml:6]: any need for the `alias/` prefix here? just curious...

File: test/e2e/tests/test_alias.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        alias_arn = cr['status']['ackResourceMetadata']['arn']
+
+        _, deleted = k8s.delete_custom_resource(ref, 3, 5)
+        assert deleted
+
+        for alias in kms_client.list_aliases(KeyId=cr['spec']['targetKeyID'])['Aliases']:
+            if alias['AliasArn'] == alias_arn:
+                # The alias was not deleted correctly
+                pytest.fail("Alias was not deleted from KMS")
+
```
Comment (jaypipes) [inline test/e2e/tests/test_alias.py:87]: nit: empty newline (also, you might want to run `black` against your Py files)

**Review by jaypipes — APPROVED** [review]
Nice, @RedbackThomson :)

File: test/e2e/resources/alias_simple.yaml (hunk 2 above)
Comment (RedbackThomson) [reply test/e2e/resources/alias_simple.yaml:6]: Server side validation requirement that all alias names begin with `alias/` - https://docs.aws.amazon.com/sdk-for-go/api/service/kms/#CreateAliasInput

File: pkg/resource/alias/sdk.go [hunk 4]
```diff
@@ -76,6 +78,18 @@ func (rm *resourceManager) sdkFind(
 	// Merge in the information we read from the API call above to the copy of
 	// the original Kubernetes object we passed to the function
 	ko := r.ko.DeepCopy()
+	// Filter resulting aliases, matching only the one with the name in the spec
+	matchingAliases := []*svcsdk.AliasListEntry{}
+	for _, elem := range resp.Aliases {
+		if elem.AliasName == nil || r.ko.Spec.Name == nil {
```
Comment (vijtrip2) [inline pkg/resource/alias/sdk.go:84]: `can r.ko.Spec.Name` check be placed outside the loop so that unnecessary iteration over resp.Aliases in not needed?

**Review by vijtrip2 — APPROVED** [review]

**Pull request merged**